### DISCUSSION TITLE
Passwords hash migration from V4 to V5

### DISCUSSION THREAD

**Discussion by DraftProducts** [adonisjs/core#1851] [Help] [answered]
Hi everyone 👋 
I have merge a full project from V4 to V5, but there is something wrong with auth.
I saw that the default hash method has change between the two versions, on v4 it was bcrypt and in the v5 the default is argon.
So i change this configuration to have back bcrypt, but something else is making the hash diffrent: the package.
It was `bcryptjs` and now `phc-bcrypt`.

Here a screen of the hash difference for two same passwords: 
![image](https://user-images.githubusercontent.com/29425008/98119920-92713400-1ead-11eb-801e-de9084b55663.png)

If someone have i idea to resolve this incompatibility, it will help me a lot !

**Comment by kmorpex** [comment]
I have the same problem

**Comment by kmorpex** [comment, marked as answer]
helped me
https://gist.github.com/thetutlage/cf32e18d66408fd96fc8b7d3591c7a1a

**Comment by thetutlage** [comment]
@DraftProducts Yes, in v5 we use PHC string format and hence old passwords cannot be migrated out of the box. However, we have plans on add legacy support with both the `bcrypt` and `argon` drivers. 

We even have this [PR open](https://github.com/adonisjs/hash/pull/4), but I am not 100% convinced wit the implementation. So if you or someone else wants to work on it, then I am happy to help.

**Reply by DraftProducts** [reply]
I will check it 👍

**Comment by josh231101** [comment]
It's been +2 years. Is there even a response on this?

**Reply by DraftProducts** [reply]
You have to make a custom hashDriver

**Reply by josh231101** [reply]
The link provided helped me! Thanks!
